### PR TITLE
Code cleanup and reorganization

### PR DESCRIPTION
**Pull request by joeosburn**

### CODE REVIEW

File: lib/encode.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+const INPUT_TABLE = require('./inputs');
+
+const SHORTHAND_FLAGS = {
+  '*': { _required: true },
+  '!': { _disabled: true },
+  '@': (model) => ({
+    _type: 'association',
+    _model: model
+  })
+  // '*': 'required',
```
Comment (shanebo) [inline lib/encode.js]: You can kill this line

File: lib/encode.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  // '@': 'model',
```
Comment (shanebo) [inline lib/encode.js]: You can kill this line

File: lib/encode.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  // ':': 'type',
+  // '!': 'pattern',
+  // '!': 'disabled'
```
Comment (shanebo) [inline lib/encode.js]: You can kill this line

File: lib/encode.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+  '*': { _required: true },
+  '!': { _disabled: true },
+  '@': (model) => ({
+    _type: 'association',
+    _model: model
+  })
+  // '*': 'required',
+  // '@': 'model',
+  // ':': 'type',
+  // '!': 'pattern',
```
Comment (shanebo) [inline lib/encode.js]: You can kill this line

File: lib/encode.js [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+  // '!': 'disabled'
+  // ':': '_input',
+  // '/': '_disabled'
```
Comment (shanebo) [inline lib/encode.js]: You can kill this line

File: lib/encode.js [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+  }
+}
+
+const toField = (prop, setting) => ({
+  ...baseField(prop),
+  ...createField(prop, setting)
+});
+
+
+const toNonMetaFields = (schema) => {
```
Comment (shanebo) [inline lib/encode.js]:
```
const toNonMetaFields = (schema) => getNonMetaProps(schema)
  .map((prop) => toField(prop, schema[prop]));
```

File: lib/encode.js (hunk 5 above)
Comment (joeosburn) [reply lib/encode.js]:
```
  // ':': 'type',
  // ':': '_input',
  // '?': '_help'
  // '()': '_placeholder'
  // '_': '_placeholder'
```

After all the updates, that's what it looks like ^

**Comment by joeosburn** [comment]
@shanebo Pushed a new commit. I'll squash if you're good with this

File: lib/fields.js [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    } else {
+      return value;
+    }
+  }).filter(phrase => phrase);
+
+  return sentence.join(' ');
+}
+
+exports.mapFieldValues = mapFieldValues;
+exports.toSentence = toSentence;
```
Comment (shanebo) [inline lib/fields.js]: @joeosburn I would put all these high level functions in the index because they're exposed by the library

**Review by shanebo — APPROVED** [review]

**Pull request merged**